### PR TITLE
add:login-page

### PR DESCRIPTION
**Pull request by alonakaliuzhna**
b34569f7dd3e472956d8c6f7be611431

### CODE REVIEW

**Comment by itsnotpolza** [comment]
add screenshots

File: src/components/LoginForm/LoginForm.jsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    reset,
+    formState: { errors }
+  } = useForm({
+    reValidateMode: 'onSubmit',
+    mode: 'onTouched'
+  });
+
+  const email = watch('email');
+  const password = watch('password');
+  console.log(watch());
```
Comment (itsnotpolza) [inline src/components/LoginForm/LoginForm.jsx:32]: remove console statement

File: src/components/LoginForm/LoginForm.jsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  const isValid = email && password;
```
Comment (itsnotpolza) [inline src/components/LoginForm/LoginForm.jsx:33]: better to have an isEmpty flag
```suggestion
  const isEmpty = !email || !password;
```

File: src/components/LoginForm/LoginForm.jsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+  const onSubmit = (data) => {
+    console.log(data);
```
Comment (itsnotpolza) [inline src/components/LoginForm/LoginForm.jsx:36]: console statement

File: src/components/LoginForm/LoginForm.jsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    reset();
+  };
+
+  const onError = (error) => {
+    console.log(error);
```
Comment (itsnotpolza) [inline src/components/LoginForm/LoginForm.jsx:41]: console

File: src/components/LoginForm/LoginForm.jsx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+          )}
+        </FormControl>
+       
+        <Button
+          type="submit"
+          disabled={!isValid}
+          mt="10px"
+          colorScheme="purple"
+        >
+          Log in
```
Comment (itsnotpolza) [inline src/components/LoginForm/LoginForm.jsx:93]: move text to i18n

File: src/pages/Login.jsx [hunk 6]
```diff
@@ -1,13 +1,17 @@
...
+
+
 export const Login = () => {
   return (
     <>
       <Box bg="bisque" w="100%" p={4} color="blueviolet">
         <Flex direction="column" justify="center" align="center">
           <main>
             <h2>{i18next.t('login.welcomeMessage')}</h2>
+    <LoginForm/>
```
Comment (itsnotpolza) [inline src/pages/Login.jsx:14]: run lint

**Comment by itsnotpolza** [comment]
Also add the page to routes

File: src/components/LoginForm/LoginForm.jsx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+            {...register('email', { required: true })}
+          />
+          {errors && errors.email && (
+            <FormHelperText color="red">
+              {errors.email.message && errors.email.message}
+            </FormHelperText>
+          )}
+        </FormControl>
+        <FormControl>
+          <FormLabel htmlFor="password">Password</FormLabel>
```
Comment (kalashnikau) [inline src/components/LoginForm/LoginForm.jsx:67]: i18n

File: src/components/LoginForm/LoginForm.jsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+    <Stack
+      width="500px"
+      p="4"
+      boxShadow="xl"
+      borderRadius="4px"
+      background="white"
+    >
+      <form onSubmit={handleSubmit(onSubmit, onError)}>
+        <FormControl>
+          <FormLabel htmlFor="email">Email</FormLabel>
```
Comment (kalashnikau) [inline src/components/LoginForm/LoginForm.jsx:54]: i18n

File: src/components/LoginForm/LoginForm.jsx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+          <Input
+            id="email"
+            type="email"
+            {...register('email', { required: true })}
+          />
+          {errors && errors.email && (
```
Comment (kalashnikau) [inline src/components/LoginForm/LoginForm.jsx:60]: OPTIONAL: maybe will be a good improvement to move the logic of getting an error message to function and then reuse it under each field.

**Review by kalashnikau — CHANGES_REQUESTED** [review]

File: src/components/LoginForm/LoginForm.jsx [hunk 10]
```diff
@@ -0,0 +1,99 @@
+import React from 'react';
```
Comment (lazzzaroni) [inline src/components/LoginForm/LoginForm.jsx:1]: We can safely remove import of React itself from our code, but it's optional.

Also I suggest to import useState hook at the top of file, not from React itself.

`import {useState} from 'react';`

File: src/components/LoginForm/LoginForm.jsx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Stack
+} from '@chakra-ui/react';
+export const LoginForm = () => {
+  const [show, setShow] = React.useState(false);
```
Comment (lazzzaroni) [inline src/components/LoginForm/LoginForm.jsx:15]: `const [show, setShow] = useState(false);`

File: src/pages/Login.jsx [hunk 12]
```diff
@@ -1,13 +1,17 @@
...
+import { LoginForm } from '@/components';
+
+
 export const Login = () => {
   return (
     <>
       <Box bg="bisque" w="100%" p={4} color="blueviolet">
         <Flex direction="column" justify="center" align="center">
           <main>
             <h2>{i18next.t('login.welcomeMessage')}</h2>
```
Comment (lazzzaroni) [inline src/pages/Login.jsx:13]: Do we need this message on login page?

File: src/components/LoginForm/LoginForm.jsx [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+  InputGroup,
+  InputRightElement,
+  Stack,
+  useColorModeValue} from '@chakra-ui/react';
+import i18next from 'i18next';
+
+import { LogoWalletIcon, UserIcon } from '@/assets';
+
+export const LoginForm = () => {
+  const [show, setShow] = useState(false);
```
Comment (kalashnikau) [inline src/components/LoginForm/LoginForm.jsx:23]: optional: probably try to use useDisclosure() from chakra-ui

File: src/components/LoginForm/LoginForm.jsx [hunk 14]
```diff
@@ -0,0 +1,131 @@
...
+        justify="center"
+        flexDirection="column"
+        borderWidth="1px"
+        borderStyle="solid"
+        borderColor="black"
+        background="white"
+      >
+        <LogoWalletIcon width="45" height="45" color={iconsThemeColor} />
+        <Heading as="h1" mr="5px" ml="5px" size="lg">
+          PortMoney
```
Comment (kalashnikau) [inline src/components/LoginForm/LoginForm.jsx:65]: move to lang variables

File: src/components/LoginForm/LoginForm.jsx [hunk 15]
```diff
@@ -0,0 +1,131 @@
...
+                {errors.email.message && errors.email.message}
+              </FormHelperText>
+            )}
+          </FormControl>
+          <FormControl>
+            <FormLabel htmlFor="password">{i18next.t('Password *')}</FormLabel>
+            <InputGroup size="md">
+              <Input
+                id="password"
+                placeholder="Password"
```
Comment (kalashnikau) [inline src/components/LoginForm/LoginForm.jsx:96]: must be lang variable

File: src/components/LoginForm/LoginForm.jsx [hunk 16]
```diff
@@ -0,0 +1,131 @@
...
+        boxShadow="xl"
+        borderRadius="4px"
+        background="white"
+      >
+        <form onSubmit={handleSubmit(onSubmit)}>
+          <FormControl>
+            <FormLabel htmlFor="email">{i18next.t('Email *')}</FormLabel>
+            <Input
+              id="email"
+              placeholder="Email"
```
Comment (kalashnikau) [inline src/components/LoginForm/LoginForm.jsx:81]: must be lang variable

**Review by kalashnikau — CHANGES_REQUESTED** [review]

File: src/components/LoginForm/LoginForm.jsx [hunk 17]
```diff
@@ -0,0 +1,134 @@
...
+  useColorModeValue
+} from '@chakra-ui/react';
+import i18next from 'i18next';
+
+import { LogoWalletIcon, UserIcon } from '@/assets';
+
+export const LoginForm = () => {
+  const [show, setShow] = useState(false);
+  const handleClick = () => setShow(!show);
+  let navigate = useNavigate();
```
Comment (itsnotpolza) [inline src/components/LoginForm/LoginForm.jsx:26]: why is this a let variable?

File: src/i18n/i18n_en.json [hunk 18]
```diff
@@ -13,5 +13,11 @@
   "header.navigation.about": "About",
   "header.navigation.categories": "Categories",
   "header.userName": "{{props.userName}}",
-  "header.userEmail": "{{props.userEmail}}"
+  "header.userEmail": "{{props.userEmail}}",
```
Comment (kalashnikau) [inline src/i18n/i18n_en.json:16]: I don't think we need to localize such fields, or I didn't get what you trying to achieve there

Comment (kalashnikau) [reply src/i18n/i18n_en.json:16]: or its even not your chnages)

File: src/components/LoginForm/LoginForm.jsx [hunk 19]
```diff
@@ -0,0 +1,131 @@
+/* eslint-disable no-unused-vars */
```
Comment (itsnotpolza) [inline src/components/LoginForm/LoginForm.jsx:1]: remove eslint-disable if there are errors - fix the errors

File: src/i18n/i18n_en.json [hunk 20]
```diff
@@ -13,5 +13,11 @@
...
+  "login.appName": "PortMoney",
+  "login.form.labelEmail": "Email",
+  "login.form.labelPassword": "Password",
+  "login.form.inputPlaceholderEmail": "Email",
+  "login.form.inputPlaceholderPassword": "Password",
```
Comment (itsnotpolza) [inline src/i18n/i18n_en.json:21]: you can reuse one key instead of creating identical ones

File: src/i18n/i18n_en.json (hunk 18 above)
Comment (alonakaliuzhna) [reply src/i18n/i18n_en.json:16]: It is not my change

**Review by itsnotpolza — APPROVED** [review]

**Review by kalashnikau — APPROVED** [review]

**Pull request merged**